mempool/mempool.go

```diff
@@ -488,8 +487,8 @@ type txCache interface {
 type mapTxCache struct {
 	mtx  sync.Mutex
 	size int
-	map_ map[string]struct{}
-	list *list.List // to remove oldest tx when cache gets too big
+	map_ map[string]*clist.CElement
+	list *clist.CList // to remove oldest tx when cache gets too big
```

Comment:
is there some reason we needed to switch to Clist?

Reply:
Yes. CheckTxs are called in parallel, and can have different execution times. I can describe problematic situations more if you'd like. (I ran through a couple in my head that I thought would eventually cause big problems before making the clist switch)

Reply:
But is there some new requirement that only now we need a clist? or you think we always needed it?

CheckTx actually don't run in parallel, they run sequentially, but responses come back asynchronously (though in order)

Reply:
Oh your right, we don't need it here. The reason we don't need it (and something I failed to consider earlier) is that we have a mutex on all deletions and additions. I forgot that we needed that for the hashmap still. I'll switch it back, thanks for pointing this out :D